transitions/core.py

```diff
@@ -359,16 +356,17 @@ def __init__(self, model=None, states=None, initial=None, transitions=None,
         self._transition_queue = deque()
         self.models = []
 
-        if initial is None:
-            self.add_states('initial')
+        if model is None and add_self:  # TODO(pbovbel) reconsider API for next major release
+            model = self
+
```

Comment:
> Actually, add_self=False and no model is exactly the use case i'm interested in

oh, I see. Well, if that is actually a use case I agree that there should be a way to achieve this.

> What kind of weird effects do your foresee?

I was thinking about the easier settings where someone tries to use `Machine` as a model but nothing happens. I am also not a big fan of attributes that depend on other attributes ('adds self but only if...') or change meaning based on the value of other attributes.

Maybe in `transitions` 0.5.x a Machine without model parameter should actually be initialized without a model. Or `add_self` should always add the machine as a model. Or we find a better way to actually pass `Machine` as a part of the model parameter which would be preferable.

> I also like the possibility of not instantiating a magic 'initial' state, **which is only possible** if no models are specified **and no initial arg is provided**.

The state 'initial' is only created if you do not pass `initial` as an argument. You could just pass `initial` to have a default initial argument for all the following models. In your scenario, you would have to pass `initial` to `add_model` anyway to not raise Exceptions. If you plan to initialize `Machine` without states this might be a problem though.